### DISCUSSION TITLE
Audio is not defined

### DISCUSSION THREAD

**Discussion by MejanH** [vercel/next.js#17963] [Help] [answered]
I tried to play an audio file when a button is triggered like the code below.

```javascript
const [audio] = useState(new Audio(URL))
audio.play()
```
but it's saying `Audio is not defined`.

**Comment by c0b41** [comment, marked as answer]
because you can't run Audio api on server

```js

const [audio, setAudio] = useState(null)

useEffect(() => {

    setAudio(new Audio(URL)) // only call client

})
```

**Reply by jamesmosier** [reply]
You will likely only want to call this once, therefore you should add a `[]` dependency array to the effect:

```js
const [audio, setAudio] = useState(null)
useEffect(() => {
  setAudio(new Audio(URL))
// only run once on the first render on the client
}, [])
```

---

Alternatively you could use [`next/dynamic` with `ssr: false`](https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr) to not render your particular component that uses the `Audio` API on the server.

**Reply by ats1999** [reply]
> Alternatively you could use next/dynamic with ssr: false to not render your particular component that uses the Audio API on the server.


No need to escape SSR, if any component have any code that can run only on the client then call it into the `useEffect` hook.
